File: exampleSources/kotlin/app1.kt
// Kotlin — data class + collection map.

data class Task(val title: String, val priority: Int, val done: Boolean)

fun main() {
    val backlog = listOf(
        Task("Write tests", 2, false),
        Task("Fix login bug", 5, false),
        Task("Refactor parser", 3, true),
        Task("Ship release", 1, false),
    )

    val open = backlog.filter { !it.done }
    val ordered = open.sortedBy { it.priority }
    val titles = ordered.map { it.title }

    titles.forEach { println(it) }
}

File: exampleSources/lua/app1.lua
-- Lua — table iteration with ipairs + custom sort.

local function by_priority(a, b)
  return a.priority < b.priority
end

local backlog = {
  { title = "Write tests",    priority = 2 },
  { title = "Fix login bug",  priority = 5 },
  { title = "Refactor parser", priority = 3 },
}

table.sort(backlog, by_priority)

for _, task in ipairs(backlog) do
  print(string.format("%d  %s", task.priority, task.title))
end